test/e2e/autoscaling_utils.go

```diff
@@ -114,14 +115,37 @@ func (rc *ConsumingRC) sendOneConsumeCPUrequest(milicores int, durationSec int)
 	expectNoError(err)
 }
 
-func (rc *ConsumingRC) CleanUp() {
+func (rc *ResourceConsumer) GetReplicas() int {
+	replicationController, err := rc.framework.Client.ReplicationControllers(rc.framework.Namespace.Name).Get(rc.name)
+	expectNoError(err)
+	if replicationController == nil {
+		Failf(rcIsNil)
+	}
+	return replicationController.Status.Replicas
+}
+
+func (rc *ResourceConsumer) ReplicasEqual(desiredReplicas int) {
+	timeout := 10 * time.Minute
+	for start := time.Now(); time.Since(start) < timeout; time.Sleep(5 * time.Second) {
```

Comment:
Querying every 5 seconds is probably too often (the whole timeout is 10 minutes). I suggest increasing this to 20 seconds.